## New Data Prediction and Accuracy

#### Try to predict data on WEC machine on another location tasmania. Which is totally new. so predict how accurate our model to predict total new data

In [86]:
from joblib import load
new_data = pd.read_csv('Tasmania_Data.csv',header=None)
new_data

0         1         2         3         4         5         6   \
0      546.1931  194.5337  160.0840  435.5412  220.8238  449.5166  396.1027   
1      281.0669  390.3761  561.0742  295.9217  565.5344  236.3035  393.0648   
2      566.0000  566.0000  346.5334  202.2120  389.6777  277.7876  421.9610   
3        2.1045  352.3969  285.2965  566.0000  532.4306  176.8103  566.0000   
4      168.5854  550.6155    0.0000  566.0000  450.8427    0.0000  566.0000   
...         ...       ...       ...       ...       ...       ...       ...   
71995  447.4090   80.6710   74.8115  495.8420  520.5531  205.9689   20.4930   
71996  565.5693  522.9972  437.3214   88.5190    8.5200  429.1224  564.0161   
71997   49.7652   90.4154  110.6000  532.7479  428.7800  101.1560   13.6231   
71998  539.4314  154.6262  493.8225    2.3327  217.3275  480.6632  565.6854   
71999  565.6854   98.0576  437.2540   58.9012  160.7589  424.0946  509.1169   

             7         8         9   ...           39           40  \
0      231.7794   69.3599  474.1572  ...  216324.3888  242794.9467   
1      340.0667  213.7463  457.8913  ...  228315.9892  196579.4315   
2      260.3339  464.1620  566.0000  ...  216919.6851  210539.3646   
3        0.0000  234.7225  485.8672  ...  231355.4783  267915.4110   
4      566.0000  458.9008  227.8836  ...  212278.2632  199341.9985   
...         ...       ...       ...  ...          ...          ...   
71995  541.6594  368.6631  529.5737  ...  246978.9773  266753.2997   
71996   38.1109  339.2581   79.0079  ...  230982.2595  215653.6026   
71997  506.2370  450.6601  552.8977  ...  222247.9781  267064.3316   
71998  165.0820   83.0391   55.2986  ...  212928.1552  174414.1646   
71999  221.6505  139.6076  111.8671  ...  225457.4783  193769.5207   

                41           42           43           44           45  \
0      234589.0464  241446.2196  266320.5629  188114.6855  170653.5874   
1      265924.2946  236228.4838  229197.3604  212649.9615  253153.4761   
2      234470.2783  245953.6945  200980.6210  203164.4624  262784.1109   
3      239599.8014  239349.6550  239027.6745  194421.2484  200775.3338   
4      230549.7763  210669.8570  271101.8480  199967.8783  203943.1047   
...            ...          ...          ...          ...          ...   
71995  212646.4644  238222.2140  265330.3691  198680.1198  205402.4175   
71996  223123.6389  185182.2772  241851.2671  265073.5264  266772.6413   
71997  257782.2912  233764.7869  241481.1899  228687.0804  210579.6182   
71998  265975.9059  265511.9281  186135.0619  212594.3104  228857.0326   
71999  267991.6425  265206.3409  230342.6521  210565.9681  230551.3241   

                46           47           48  
0      195915.6535  180922.3979  3648666.894  
1      265363.3600  251322.8751  3810742.386  
2      269371.6293  265446.7199  3745925.666  
3      265827.1867  271780.8771  3781329.415  
4      265087.0171  256492.7005  3667609.449  
...            ...          ...          ...  
71995  163774.3168  141090.0287  3536650.863  
71996  200968.4820  230855.6899  3645856.545  
71997  236459.0060  222454.7973  3718330.844  
71998  267671.1880  249700.4554  3685475.541  
71999  265369.2457  240729.0166  3733281.103  

[72000 rows x 49 columns]

In [87]:
# Separate features and target
X_new = new_data.iloc[:, :-1]
y_true = new_data.iloc[:, -1]

In [88]:
# Load the saved scaler
scaler = load("scaler.joblib")
X_new_scaled = scaler.transform(X_new)

In [89]:
# Load the saved best model
best_saved_model= load("Best_Model_SVM_Linear_Kernel.joblib")
best_saved_model

SVR(kernel='linear')

In [91]:
# Predict using the loaded model
y_pred = best_saved_model.predict(X_new_scaled)

In [92]:
# Evaluate the performance
rmse = mean_squared_error(y_true, y_pred)
r2 = r2_score(y_true, y_pred)

# Print evaluation results
print(f"RMSE on New Data: {rmse:.4f}")
print(f"R^2 Score on New Data: {r2:.4f}")

# Save predictions alongside true values for further analysis
results = pd.DataFrame({
    "Actual": y_true,
    "Predicted": y_pred
})

RMSE on New Data: 19177.6745
R^2 Score on New Data: 1.0000


In [93]:
results.to_csv("new_data_predictions.csv", index=False)
print("Predictions saved to 'new_data_predictions.csv'.")

Predictions saved to 'new_data_predictions.csv'.


**Observation on Best Model's Performance on New Data (Tasmania)**
Performance Metrics: RMSE: 19,177.67, R²: 1.0000

1. **Interpretation of R² Score:** The 𝑅2 score of 1.0000 indicates that the model perfectly explains the variance in the target variable (Powerall) for the new dataset. This is an exceptional result, suggesting the model is highly effective in capturing the underlying patterns and relationships in the data.

2. **RMSE Interpretation:** The RMSE of 19,177.67 represents the average prediction error in the same units as the target variable (Powerall). While the RMSE value might appear high, it must be compared to the scale of Powerall in the new dataset. If Powerall values are in a similar range, this RMSE could indicate a very accurate prediction.

3. **Consistency Across Datasets:** The performance on the new dataset (Tasmania) aligns with the model's exceptional results on the previous datasets. This consistency demonstrates that the model generalizes well to unseen data, reinforcing its reliability and robustness.

4. **Insights on Data and Model:** The high 𝑅2 score and acceptable RMSE suggest that the placement of WECs and their features in Tasmania align with patterns learned from the Adelaide dataset. The SVM Linear Kernel model effectively captures these relationships, making it highly suitable for predicting Powerall in similar contexts.

5. **Model Robustness:** Achieving such a high 𝑅2 score on a completely new dataset (Tasmania) indicates the model's robustness. It shows the model's ability to generalize its learning across different wave energy converter configurations and power outputs.

# The End